Wikipedia/Code/WMFCVLInfo.m

```diff
@@ -0,0 +1,337 @@
+            if (headerAttributes != nil) {
+                headerAttributes.frame = CGRectMake(x, y, columnWidth, headerHeight);
+                [section addHeader:headerAttributes];
+            }
+            [invalidatedHeaderIndexPaths addObject:supplementaryViewIndexPath];
+        } else {
+            WMFCVLAttributes *headerAttributes = section.headers[0];
+            headerHeight = headerAttributes.frame.size.height;
+            CGRect headerFrame =  CGRectMake(x, y, columnWidth, headerHeight);
+            if (!CGRectEqualToRect(headerFrame, headerAttributes.frame)) {
+                WMFCVLAttributes *newAttributes = [headerAttributes copy];
+                newAttributes.frame = headerFrame;
+                [section replaceHeaderAtIndex:0 withHeader:newAttributes];
+                [invalidatedHeaderIndexPaths addObject:supplementaryViewIndexPath];
+            }
+        }
+        
+        sectionHeight += headerHeight;
+        y += headerHeight;
+        
+        CGFloat itemX = x + sectionInsets.left;
+        CGFloat itemWidth = columnWidth - sectionInsets.left - sectionInsets.right;
+        for (NSInteger item = 0; item < [collectionView.dataSource collectionView:collectionView numberOfItemsInSection:sectionIndex]; item++) {
+            if (item == 0) {
+                y += sectionInsets.top;
+            } else {
+                y += interItemSpacing;
+            }
+            
+            CGFloat itemHeight = 0;
+            NSIndexPath *itemIndexPath = [NSIndexPath indexPathForItem:item inSection:sectionIndex];
+            
+            if (item >= section.items.count) {
+                itemHeight = [delegate collectionView:collectionView estimatedHeightForItemAtIndexPath:[NSIndexPath indexPathForItem:item inSection:sectionIndex] forColumnWidth:columnWidth];
+                
+                
+                [invalidatedItemIndexPaths addObject:itemIndexPath];
+                WMFCVLAttributes *itemAttributes = (WMFCVLAttributes *)[WMFCVLAttributes layoutAttributesForCellWithIndexPath:itemIndexPath];
+                if (itemAttributes != nil) {
+                    itemAttributes.frame = CGRectMake(itemX, y, itemWidth, itemHeight);
+                    [section addItem:itemAttributes];
+                }
+            } else {
+                WMFCVLAttributes *itemAttributes = section.items[item];
+                itemHeight = itemAttributes.frame.size.height;
+                CGRect itemFrame = CGRectMake(itemX, y, itemWidth, itemHeight);
+                if (!CGRectEqualToRect(itemFrame, itemAttributes.frame)) {
+                    WMFCVLAttributes *newAttributes = [itemAttributes copy];
+                    newAttributes.frame = itemFrame;
+                    [section replaceItemAtIndex:item withItem:newAttributes];
+                    [invalidatedItemIndexPaths addObject:itemIndexPath];
+                }
+            }
+            
+            assert(itemHeight > 0);
```

Comment:
NSAssert can make it clearer to someone using the code later the reason for the assertion failure because it takes a message. Also it throws an exception rather than just crashing.